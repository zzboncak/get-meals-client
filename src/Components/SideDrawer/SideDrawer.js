import React from 'react';
import { NavLink } from "react-router-dom";
import './SideDrawer.css';


const sideDrawer = props => {
  let drawerClasses = 'side-drawer';
  if (props.show) {
    drawerClasses = 'side-drawer open';
  }
  return (
    <nav className={drawerClasses}>
      <ul>
        <li><NavLink to="/" onClick={props.drawerClickHandler}>Home</NavLink></li>
        <li><NavLink to="/add-location" onClick={props.drawerClickHandler}>Add Location</NavLink></li>
        <li><NavLink to="/About" onClick={props.drawerClickHandler}>About Us</NavLink></li>
      </ul>
    </nav>
);
};

export default sideDrawer;