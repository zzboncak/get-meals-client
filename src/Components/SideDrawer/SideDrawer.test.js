import React from 'react';
import ReactDOM from 'react-dom';
import SideDrawer from './SideDrawer';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe(`SideDrawer component`, () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BrowserRouter>
        <SideDrawer />
    </BrowserRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <SideDrawer />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
