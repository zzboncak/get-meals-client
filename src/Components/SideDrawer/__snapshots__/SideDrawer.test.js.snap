// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideDrawer component renders the component as expected 1`] = `
<nav
  className="side-drawer"
>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/add-location"
        onClick={[Function]}
      >
        Add Location
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/About"
        onClick={[Function]}
      >
        About Us
      </a>
    </li>
  </ul>
</nav>
`;
